### PR TITLE
Use page language code to select service language override

### LINKED ISSUE #7258: Set service language override more inclusively
### Is your feature request related to a problem? Please describe.
Some Services, e.g. Ukrainian, publish STYs in multiple languages. In order to support this, the page needs to have access to the relevant service information, like translations.

### Describe the solution you'd like
- Currently the Ukraine in Russian index makes the Russian configuration available by setting a language override (or variant) that matches a key in `src/app/lib/config/services/ukrainian.js` if the pathname equals `/ukrainian/ukraine_in_russian`. This will not work for STYs.
- We can make use of `pageData.metadata.language` to detect what language the current page is written in.

### Describe alternatives you've considered
n/a

### Testing notes
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

### Additional context
- [langHandler](https://github.com/bbc/simorgh/blob/722b42256ba0fb8585914a418176bc539562bccd/src/app/lib/utilities/langHandler/index.js)

### PR DESCRIPTION
**Pull request by simonsinclair**
Resolves https://github.com/bbc/simorgh/issues/7258

### Overall change

Use the page language code instead of a fixed pathname to select the service language override.

### Code changes

- Pass the page language code instead of pathname to `ServiceContextProvider`.
- Update `getLangOverride` to select the language override based on the page language code.
- Update `langHandler` unit tests.
- Update `ServiceContext` unit tests.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

### Testing

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/langHandler/index.test.js [hunk 1]
```diff
@@ -1,18 +1,11 @@
...
-    ).toEqual('ru-UA');
+  it('should return the name of a service lang override for a given page lang', () => {
+    expect(getLangOverride('ru')).toEqual('ru-UA');
   });
 
-  it('should return null if pathname is incorrect', () => {
-    expect(getLangOverride({ pathname: '/ukrainian/not_russian' })).toEqual(
-      null,
-    );
+  it('should return undefined if the given page lang does not have a service lang override', () => {
```
Comment (HarveyPeachey) [inline src/app/lib/utilities/langHandler/index.test.js:8]: Potentially a test case for `undefined`? Seeing as we can now pass an `undefined` lang from this https://github.com/bbc/simorgh/blob/f5ba5a206c906c2826060cb6cde6a47c795c5f16/src/app/containers/PageHandlers/withContexts.jsx#L35

**Review by HarveyPeachey — COMMENTED** [review]
Looks good! Just a potential test case

File: src/app/lib/utilities/langHandler/index.test.js (hunk 1 above)
Comment (simonsinclair) [reply src/app/lib/utilities/langHandler/index.test.js:8]: Thanks, Harvey. I can see how it'd be helpful to see an assertion for this – added.

**Review by HarveyPeachey — APPROVED** [review]
LGTM

**Review by chris-hinds — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
Chat with Simon, adding here so that it will be useful for similar testing in future:

* pageData.metadata.language comes from Aresand is a two letter code that is transformed into a 4 letter code in langHandler which then matches https://github.com/bbc/simorgh/blob/latest/src/app/lib/config/services/ukrainian.js#L279

* https://ares-api.test.api.bbci.co.uk/api/asset/ukrainian/23154182 --> Ares feed

Looks good to me..

**Pull request merged**